gensim/models/doc2vec.py

```diff
@@ -794,6 +794,23 @@ def train(self, documents=None, corpus_file=None, total_examples=None, total_wor
 
         """
         kwargs = {}
+
+        # Check if both documents and corpus_file are None
+        if corpus_file is None and documents is None:
+            raise TypeError("Either one of corpus_file or documents value must be provided.")
```

Comment:
```suggestion
            raise TypeError("Either one of corpus_file or documents value must be provided")
```

I tried committing these nitpicks to your branch myself, but I don't have permissions.